File: summit_harness/evidence.py
import json
from pathlib import Path
from typing import Any, Dict, List


class EvidenceWriter:
    def __init__(self, output_dir: str):
        self.output_dir = Path(output_dir)
        self.output_dir.mkdir(parents=True, exist_ok=True)

    def write(self, report: dict[str, Any], metrics: dict[str, Any]):
        # Write report.json
        with open(self.output_dir / "report.json", "w", encoding="utf-8") as f:
            json.dump(report, f, sort_keys=True, indent=2)

        # Write metrics.json
        with open(self.output_dir / "metrics.json", "w", encoding="utf-8") as f:
            json.dump(metrics, f, sort_keys=True, indent=2)

        # Write index.json
        # The key should be the evidence ID, and it should contain a "files" list.
        primary_evd_id = report.get("evidence_id")
        if not primary_evd_id and report.get("evidence_ids"):
            primary_evd_id = report["evidence_ids"][0]

        if not primary_evd_id:
            primary_evd_id = f"EVD-UNKNOWN-{report.get('run_id', 'unknown')}"

        index = {
            "version": "1.0",
            "items": {
                primary_evd_id: {
                    "files": [
                        str(self.output_dir / "report.json"),
                        str(self.output_dir / "metrics.json"),
                        str(self.output_dir / "stamp.json")
                    ]
                }
            }
        }
        with open(self.output_dir / "index.json", "w", encoding="utf-8") as f:
            json.dump(index, f, sort_keys=True, indent=2)

    def write_stamp(self, stamp: dict[str, Any]):
        with open(self.output_dir / "stamp.json", "w", encoding="utf-8") as f:
            json.dump(stamp, f, sort_keys=True, indent=2)

File: swarm/evidence.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.emitEvidence = emitEvidence;
const fs_1 = __importDefault(require("fs"));
const path_1 = __importDefault(require("path"));
function emitEvidence(runId, plan, result, baseDir = 'evidence/runs') {
    const runDir = path_1.default.join(baseDir, runId);
    if (!fs_1.default.existsSync(runDir)) {
        fs_1.default.mkdirSync(runDir, { recursive: true });
    }
    const reportPath = path_1.default.join(runDir, 'report.json');
    const metricsPath = path_1.default.join(runDir, 'metrics.json');
    const stampPath = path_1.default.join(runDir, 'stamp.json');
    const report = { run_id: runId, mode: "swarm", summary: "Swarm execution completed", evidence_ids: [`EVD-KIMIK25-SWARM-RUN-${runId}`], outputs: result.outputs };
    const metrics = { run_id: runId, metrics: result.usage };
    const stamp = { run_id: runId, generated_at: new Date().toISOString() };
    fs_1.default.writeFileSync(reportPath, JSON.stringify(report, null, 2));
    fs_1.default.writeFileSync(metricsPath, JSON.stringify(metrics, null, 2));
    fs_1.default.writeFileSync(stampPath, JSON.stringify(stamp, null, 2));
    updateIndex(`EVD-KIMIK25-SWARM-RUN-${runId}`, [reportPath, metricsPath, stampPath]);
    return runDir;
}
function updateIndex(evidenceId, files) {
    const indexPath = path_1.default.join(process.cwd(), 'evidence/index.json');
    if (fs_1.default.existsSync(indexPath)) {
        const index = JSON.parse(fs_1.default.readFileSync(indexPath, 'utf-8'));
        index.items[evidenceId] = { files: files, title: `Swarm Run ${evidenceId.slice(-8)}`, category: "runtime" };
        fs_1.default.writeFileSync(indexPath, JSON.stringify(index, null, 2));
    }
}
